serverless.yml

```diff
@@ -0,0 +1,126 @@
+service: lp-msarchitect-aws-store-fe
+frameworkVersion: '2'
+
+plugins:
+  - serverless-finch
+  - serverless-single-page-app-plugin
+
+provider:
+  name: aws
+  runtime: nodejs12.x
+  region: eu-west-1
+
+custom:
+  ## Serverless-finch condiguration
+  client:
+    bucketName: lp-msarchitect-s3-bucket
+    distributionFolder: build
+
+  ## Serverless-single-page-app-plugin configuration
+  s3LocalPath: ${self:custom.client.distributionFolder}/
+  s3BucketName: ${self:custom.client.bucketName}
+
+resources:
+  Resources:
+    ## S3 Bucket
+    WebAppS3Bucket:
+      Type: AWS::S3::Bucket
+      Properties:
+        BucketName: ${self:custom.s3BucketName}
+        AccessControl: PublicRead
+        WebsiteConfiguration:
+          IndexDocument: index.html
+          ErrorDocument: index.html
+
+    WebAppS3BucketPolicy:
+      Type: 'AWS::S3::BucketPolicy'
+      Properties:
+        Bucket:
+          Ref: WebAppS3Bucket
+        PolicyDocument:
+          Statement:
+            - Sid: 'AllowCloudFrontAccessIdentity'
+              Effect: Allow
+              Action: s3:GetObject
+              Resource: !Join
+                - ''
+                - - 'arn:aws:s3:::'
+                  - !Ref WebAppS3Bucket
+                  - /*
+              Principal:
+                AWS: !Join
+                  - ' '
+                  - - 'arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity'
+                    - !Ref OriginAccessIdentity
+
+    OriginAccessIdentity:
+      Type: AWS::CloudFront::CloudFrontOriginAccessIdentity
+      Properties:
+        CloudFrontOriginAccessIdentityConfig:
+          Comment: Access identity between CloudFront and S3 bucket
+
+    ## CoudFront Distribution 
+    WebAppCloudFrontDistribution:
+      Type: AWS::CloudFront::Distribution
+      Properties:
+        DistributionConfig:
+          Origins:
+            - DomainName: ${self:custom.s3BucketName}.s3.amazonaws.com
+              ## An identifier for the origin which must be unique within the distribution
+              Id: myS3Origin
+              ## In case you don't want to restrict the bucket access use CustomOriginConfig and remove S3OriginConfig
+              S3OriginConfig:
+                OriginAccessIdentity: !Sub origin-access-identity/cloudfront/${OriginAccessIdentity}
+              # CustomOriginConfig:
```

Comment:
You should configure `OriginShield` with the `OriginShieldRegion` here in order to maintain access to the S3 bucket in the eu-west-1 region.